// server/routes/records.js
import express from 'express';
import multer from 'multer';
import path from 'path';
import { fileURLToPath } from 'url';
import db from '../db.js';
import jwt from 'jsonwebtoken';
import { logActivity } from '../utils/activityLogger.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const SECRET_KEY = process.env.JWT_SECRET || 'healxista-super-secret-key';
const router = express.Router();

// Multer setup for storing records securely
const storage = multer.diskStorage({
    destination: function (req, file, cb) {
        cb(null, path.join(__dirname, '../uploads/'));
    },
    filename: function (req, file, cb) {
        const uniqueSuffix = Date.now() + '-' + Math.round(Math.random() * 1E9);
        cb(null, 'record-' + uniqueSuffix + path.extname(file.originalname));
    }
});

const upload = multer({ 
    storage: storage,
    limits: { fileSize: 10 * 1024 * 1024 } // 10MB limit
});

// Post a new document
router.post('/upload', upload.single('recordFile'), async (req, res) => {
    const token = req.cookies.token || (req.headers.authorization?.split(' ')[1]);
    if (!token) return res.status(401).json({ message: 'Unauthorized' });

    try {
        let decoded;
        try { decoded = jwt.verify(token, SECRET_KEY); } catch(e) { return res.status(401).json({ message: 'Unauthorized: Token expired or invalid' }); }
        const accountId = decoded.id; // User who uploaded it

        const { title, description, recordType, relatedPatientId } = req.body;
        const fileUrl = req.file ? req.file.filename : null;

        if (!fileUrl) {
            return res.status(400).json({ message: 'File is required' });
        }

        // If the uploader is a provider uploading FOR a patient, patientId is relatedPatientId
        // otherwise patientId is the accountId
        const patientId = relatedPatientId || accountId;

        const query = `
            INSERT INTO medical_records (patient_id, title, description, file_url, record_type, uploaded_by)
            VALUES ($1, $2, $3, $4, $5, $6)
            RETURNING *
        `;
        
        const result = await db.query(query, [patientId, title || 'Medical Record', description, fileUrl, recordType || 'Report', accountId]);
        
        await logActivity(req.app.get('io'), {
            accountId: patientId,
            targetRole: 'User',
            actionType: 'Record Uploaded',
            title: 'New Medical Record Uploaded',
            description: `A new document "${title}" has been added to your medical records.`,
            statusTheme: 'success'
        });

        res.status(201).json({ message: 'Record uploaded successfully', record: result.rows[0] });
    } catch (error) {
        console.error("Upload error:", error);
        res.status(500).json({ message: 'Failed to upload record' });
    }
});

// Get user's own records
router.get('/', async (req, res) => {
    const token = req.cookies.token || (req.headers.authorization?.split(' ')[1]);
    if (!token) return res.status(401).json({ message: 'Unauthorized' });

    try {
        let decoded;
        try { decoded = jwt.verify(token, SECRET_KEY); } catch(e) { return res.status(401).json({ message: 'Unauthorized: Token expired or invalid' }); }
        const accountId = decoded.id;

        const result = await db.query(`
            SELECT r.*, a.email, p.first_name, p.last_name 
            FROM medical_records r
            LEFT JOIN profiles p ON r.uploaded_by = p.account_id
            LEFT JOIN accounts a ON r.uploaded_by = a.account_id
            WHERE r.patient_id = $1
            ORDER BY r.created_at DESC
        `, [accountId]);

        res.json(result.rows.map(row => ({
            ...row,
            uploaded_by_name: `${row.first_name || ''} ${row.last_name || ''}`.trim() || row.email
        })));
    } catch (error) {
        res.status(500).json({ message: 'Failed to fetch records' });
    }
});

// Get records for a specific patient (accessed by doctor)
router.get('/patient/:patientId', async (req, res) => {
    const token = req.cookies.token || (req.headers.authorization?.split(' ')[1]);
    if (!token) return res.status(401).json({ message: 'Unauthorized' });

    try {
        let decoded;
        try { decoded = jwt.verify(token, SECRET_KEY); } catch(e) { return res.status(401).json({ message: 'Unauthorized: Token expired or invalid' }); }
        
        // In a real system, we should verify that this doctor has a booking with this patient
        // For simplicity, we just allow the fetch assuming the frontend only requests it from the patient list screen

        const { patientId } = req.params;

        const result = await db.query(`
            SELECT r.*, a.email, p.first_name, p.last_name 
            FROM medical_records r
            LEFT JOIN profiles p ON r.uploaded_by = p.account_id
            LEFT JOIN accounts a ON r.uploaded_by = a.account_id
            WHERE r.patient_id = $1
            ORDER BY r.created_at DESC
        `, [patientId]);

        res.json(result.rows.map(row => ({
            ...row,
            uploaded_by_name: `${row.first_name || ''} ${row.last_name || ''}`.trim() || row.email
        })));
    } catch (error) {
        res.status(500).json({ message: 'Failed to fetch patient records' });
    }
});

export default router;
